### PR TITLE
Add name ordering for streams and map based storages

### PR DESCRIPTION
**Pull request by carlesarnal**

### CODE REVIEW

File: app/src/main/java/io/apicurio/registry/util/SearchUtil.java [hunk 1]
```diff
@@ -29,16 +29,30 @@
...
 
-    public static int compare(SortOrder sortOrder, String id1, String id2) {
+    public static int compare(SortOrder sortOrder, ArtifactMetaDataDto id1, ArtifactMetaDataDto id2) {
         switch (sortOrder) {
             case desc:
-                return id2.compareToIgnoreCase(id1);
+                return id2.getName().compareToIgnoreCase(id1.getName());
             default:
-                return id1.compareToIgnoreCase(id2);
+                return id1.getName().compareToIgnoreCase(id2.getName());
```
Comment (EricWittmann) [inline app/src/main/java/io/apicurio/registry/util/SearchUtil.java:55]: Name could be null for either or both.

File: app/src/test/java/io/apicurio/registry/RegistryClientTest.java [hunk 2]
```diff
@@ -188,6 +188,61 @@ public void testLabels(Supplier<RegistryService> supplier) throws Exception {
...
+                    ("{\"type\":\"record\",\"title\":\"" + secondName + "\",\"fields\":[{\"name\":\"foo\",\"type\":\"string\"}]}")
+                            .getBytes(StandardCharsets.UTF_8));
+
+            CompletionStage<ArtifactMetaData> secondCs = client.createArtifact(ArtifactType.JSON, secondArtifactId, null, secondData);
+            long secondId = ConcurrentUtil.result(secondCs).getGlobalId();
 
+            retry(() -> {
+                ArtifactMetaData artifactMetaData = client.getArtifactMetaDataByGlobalId(secondId);
+                Assertions.assertNotNull(artifactMetaData);
+            });
```
Comment (EricWittmann) [inline app/src/test/java/io/apicurio/registry/RegistryClientTest.java:227]: Also maybe throw in an artifact with no name.  In that case the *ID* should be used as the name.

**Comment by carlesarnal** [comment]
Ok, I forgotten that the name can be null, agree that there should be only one comparator.

**Comment by EricWittmann** [comment]
Had a bug in it but fixed now and merged

**Pull request merged**